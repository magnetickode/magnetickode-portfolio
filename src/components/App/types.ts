import { MutableRefObject } from "react";

export interface Props {
  navbarRef: MutableRefObject<HTMLDivElement | null>;
}
