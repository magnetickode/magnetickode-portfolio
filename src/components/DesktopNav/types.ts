export interface Props {
  hashRoute: string;
}

export interface StyledNavLinkProps {
  active: boolean;
}
