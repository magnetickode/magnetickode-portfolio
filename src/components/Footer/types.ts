export interface Props {
  footerState: string;
  footerPosition: string;
}

export interface StyledFooterProps {
  footerState: string;
  footerPosition: string;
}
