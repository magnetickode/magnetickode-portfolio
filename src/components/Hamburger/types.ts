export interface Props {
  opened: boolean;
  toggleHamburger: () => void;
}

export interface StyledHamburgerProps {
  opened: boolean;
}
