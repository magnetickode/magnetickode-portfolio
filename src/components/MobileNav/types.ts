export interface Props {
  collapse: boolean;
  hashRoute: string;
}

export interface StyledMobileNavProps {
  collapse: boolean;
}

export interface StyledNavLinkProps {
  active: boolean;
}
