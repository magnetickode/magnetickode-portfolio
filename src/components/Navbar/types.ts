import { ChangeThemeGen } from "../../store/actions/types";

export interface Props {
  hamburgerOpened: boolean;
  mobileNavVisible: boolean;
  toggleHamburger: () => void;
  isDesktop: boolean;
  stickyNavVisible: boolean;
  hashRoute: string;
  changeTheme: ChangeThemeGen;
}

export interface StyledNavbarProps {
  sticky: boolean;
}
