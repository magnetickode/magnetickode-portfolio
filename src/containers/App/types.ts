import {
  ScreenSizeChangedGen,
  ShowStickyNavGen,
  ChangeScrollGen,
  ChangeHashRouteGen
} from "../../store/actions/types";

export interface Props {
  isDesktop: boolean;
  themeIndex: number;
  stickyNavVisible: boolean;
  scrolled: boolean;
  hashRoute: string;
  screenSizeChanged: ScreenSizeChangedGen;
  showStickyNav: ShowStickyNavGen;
  changeScroll: ChangeScrollGen;
  changeHashRoute: ChangeHashRouteGen;
}
