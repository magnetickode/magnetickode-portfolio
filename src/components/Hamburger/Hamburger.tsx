import React from "react";
import styled from "styled-components";

import { Props, StyledHamburgerProps } from "./types";

const StyledHamburger = styled.div<StyledHamburgerProps>`
  width: 4rem;
  transform: scale(0.8);
  cursor: pointer;

  :after,
  :before,
  & > div {
    background-color: ${({ theme }) => theme.textColor};
    border-radius: 3px;
    content: "";
    display: block;
    height: 5px;
    margin: 7px 0;
    transition: all 0.2s ease-in-out;
  }

  ${({ opened }) =>
    opened &&
    `
    :before {
      transform: translateY(12px) rotate(135deg);
    }

    :after {
      transform: translateY(-12px) rotate(-135deg);
    }

    & > div {
      transform: scale(0);
    }
  `}
`;

const Hamburger = ({ opened, toggleHamburger }: Props) => (
  <StyledHamburger
    opened={opened}
    onClick={toggleHamburger}
    data-test="HamburgerComponent"
  >
    <div />
  </StyledHamburger>
);

export default Hamburger;
