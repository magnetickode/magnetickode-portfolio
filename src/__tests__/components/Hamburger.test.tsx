import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import Hamburger from "../../components/Hamburger/Hamburger";
import { Props } from "../../components/Hamburger/types";

let wrapper: ShallowWrapper;

const props: Props = {
  opened: false,
  toggleHamburger: () => null
};

beforeEach(() => {
  wrapper = shallow(<Hamburger {...props} />);
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("renders without error", () => {
  const HamburgerComponent = wrapper.find("[data-test='HamburgerComponent']");
  expect(HamburgerComponent.length).toBe(1);
});
