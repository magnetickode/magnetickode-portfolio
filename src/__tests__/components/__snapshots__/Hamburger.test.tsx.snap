// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.div
  data-test="HamburgerComponent"
  onClick={[Function]}
  opened={false}
>
  <div />
</styled.div>
`;
